cerberus/views/invoice.py

```diff
@@ -0,0 +1,128 @@
+
+
+class InvoiceCRUD(CRUDViews):
+    model = Invoice
+    form_class = InvoiceForm
+    filter_class = InvoiceFilter
+    sortable_fields = ["id", "customer", "total"]
+
+    @classmethod
+    def get_view_class(cls, action: Actions):
+        match action:
+            case Actions.UPDATE:
+                return InvoiceUpdateView
+            case Actions.CREATE:
+                return InvoiceCreateView
+            case _:
+                return super().get_view_class(action)
+
+    @extra_view(detail=True, methods=["get", "post"])
+    def email(self: Self, request: HttpRequest, pk: int) -> HttpResponse:
+        invoice = get_object_or_404(Invoice, pk=pk)
+        if request.method != "POST":
+            return render(request, "cerberus/invoice_email_confirm.html", {"object": invoice, "invoice": invoice})
+        try:
+            invoice.resend_email()
+        except AssertionError as e:
+            return HttpResponseNotAllowed(f"Email not sent: {e}")
+
+        return render(request, "cerberus/invoice_email_sent.html", {"object": invoice, "invoice": invoice})
+
+    @extra_view(detail=True, methods=["get"], url_name="invoice_pdf")
+    def download(self: Self, request: HttpRequest, pk: int) -> HttpResponse:
+        invoice: Invoice = get_object_or_404(Invoice, pk=pk)
+
+        return invoice.get_pdf_response()
+
+    @extra_view(detail=False, methods=["get", "post"], url_name="invoice_from_charges")
+    def from_charges(self: Self, request: HttpRequest) -> HttpResponse:
+        charge_form = UninvoicedChargesForm(request.POST)
+
+        if charge_form.is_valid():
+            with transaction.atomic():
+                invoice = Invoice.from_customer(charge_form.cleaned_data["customer"])
+                for charge in charge_form.cleaned_data["charges"]:
+                    charge.invoice = invoice
+                    charge.save()
+
+            success_url = reverse_lazy("invoice_detail", kwargs={"pk": invoice.pk})
+            if request.htmx:  # type: ignore
+                return HttpResponseClientRedirect(success_url)
+            return HttpResponseRedirect(success_url)
+
+        return render(request, "cerberus/customer_charges.html", {"form": charge_form})
+
+    @extra_view(detail=True, methods=["get", "post"], url_path="send", url_name="invoice_send")
+    def send_form(self, request, pk):
+        invoice = get_object_or_404(Invoice, pk=pk)
+        inital = {"to": getattr(invoice.customer, "invoice_email", ""), "send_email": True}
+        form = InvoiceSendForm(request.POST or inital)
+
+        if request.POST and form.is_valid():
+            invoice.send(form.cleaned_data["to"], form.cleaned_data["send_email"])
+            return HttpResponseRedirect(reverse_lazy("invoice_detail", kwargs={"pk": invoice.pk}))
+
+        return render(request, "cerberus/invoice_send_form.html", {"invoice": invoice, "form": form})
+
+
+class InvoiceActionsView(TransitionView):
+    model = Invoice
+    field = "state"
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **question (code_refinement):** Review the necessity of InvoiceActionsView inheriting from TransitionView.

Ensure that the inheritance from TransitionView is necessary and that the functionality aligns with the intended use case for InvoiceActionsView.